### PR TITLE
MM-55621 fix layout for tablets and foldables

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Fixes the layout for Android tablets and foldables assigning the correct values for isTablet and isSplitView by following this guide https://developer.android.com/guide/topics/large-screens/support-different-screen-sizes

also moved the native function to be synchronous instead of async

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55621

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: android/app/src/main/java/com/mattermost/rnbeta/FoldableObserver.kt [hunk 1]
```diff
@@ -25,27 +47,40 @@ class FoldableObserver(private val activity: Activity) {
...
-                            splitViewModule?.setDeviceFolded(false)
-                        isTableTopPosture(foldingFeature) ->
-                            splitViewModule?.setDeviceFolded(false)
-                        isBookPosture(foldingFeature) ->
-                            splitViewModule?.setDeviceFolded(false)
-                        else -> {
-                            splitViewModule?.setDeviceFolded(true)
-                        }
-                    }
+                    isFolded(layoutInfo)
```
[the remark this answers is not in the document]
Comment (enahum) [reply android/app/src/main/java/com/mattermost/rnbeta/FoldableObserver.kt:50]: in fact the function returns a boolean so is ok to be called `isFolded`

File: app/constants/device.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
-import DeviceInfo from 'react-native-device-info';
 import FileSystem from 'react-native-fs';
 
+import {isTablet} from '@utils/helpers';
+
 export default {
     DOCUMENTS_PATH: `${FileSystem.CachesDirectoryPath}/Documents`,
-    IS_TABLET: DeviceInfo.isTablet(),
+    IS_TABLET: isTablet(),
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/constants/device.ts:10]: There are two things that depend on this still.
1. A component from calls that for some reason unknown to me is not using `useIsTablet` perhaps @cpoile can share more light on that
2. The markdown table component which is a large class component that I did not want to refactor in this PR

File: app/hooks/device.ts [hunk 3]
```diff
@@ -2,37 +2,15 @@
...
-import {Device} from '@constants';
+import {DeviceContext} from '@context/device';
 
 import type {KeyboardTrackingViewRef} from 'react-native-keyboard-tracking-view';
 
-const {SplitView} = NativeModules;
-const {isRunningInSplitView} = SplitView;
-const emitter = new NativeEventEmitter(SplitView);
-
 export function useSplitView() {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/hooks/device.ts:12]: I should remove the other ones that are not being used and forgot to remove

**Review by larkox — APPROVED** [review]

File: app/constants/device.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (cpoile) [reply app/constants/device.ts:10]: I can't find where `IS_TABLET` is used as a constant in the calls code, maybe I'm missing it? Please let me know where it is and for sure I'll change it if we can.

**Review by cpoile — APPROVED** [review]
Thank you, looks great (more work than I would have thought...)

File: app/constants/device.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (enahum) [reply app/constants/device.ts:10]: https://github.com/mattermost/mattermost-mobile/blob/main/app/products/calls/components/audio_device_button.tsx#L35

**Pull request merged**